parsl/app/futures.py

```diff
@@ -56,8 +72,18 @@ def __init__(self, fut: Future, file_obj: File, tid: int) -> None:
         self._tid = tid
         self.file_obj = file_obj
         self.parent = fut
-
+        if app_fut:
```

Comment:
see other comment about app_fut always being `None` - if so, this `if` doesn't need to exist.

Reply:
There is a least one instance where it is (see above comment)